webhook_to_fedora_messaging/models/util.py

```diff
@@ -0,0 +1,64 @@
+# SPDX-FileCopyrightText: Contributors to the Fedora Project
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import datetime
+
+from sqlalchemy import Column, UnicodeText
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.expression import FunctionElement
+from sqlalchemy.types import DateTime as SQLDateTime
+from sqlalchemy.types import TypeDecorator
+
+from uuid import uuid4
+
+
+class utcnow(FunctionElement):
+    """
+    Current timestamp in UTC for SQL expressions
+    """
+    type = SQLDateTime
+    inherit_cache = True
+
+
+class TZDateTime(TypeDecorator):
+    """
+    The TZDateTime class is taken from
+    https://docs.sqlalchemy.org/en/14/core/custom_types.html#store-timezone-aware-timestamps-as-timezone-naive-utc
+    """
+
+    impl = SQLDateTime
+    cache_ok = True
+
+    def process_bind_param(self, value, dialect):
+        if value is not None:
+            if not value.tzinfo:
+                raise TypeError("tzinfo is required")
+            value = value.astimezone(datetime.timezone.utc).replace(tzinfo=None)
+        return value
+
+    def process_result_value(self, value, dialect):
+        if value is not None:
+            value = value.replace(tzinfo=datetime.timezone.utc)
+        return value
+
+
+@compiles(utcnow, "postgresql")
+def _postgresql_utcnow(element, compiler, **kwargs):
+    return "(NOW() AT TIME ZONE 'utc')"
+
+
+class UUIDCreatableMixin:
+    """
```

Comment:
Why do we want that? How do you expect it to be used?

Reply:
URL paths?

Suppose a certain service is named `My Fancy GitHub Webhook` service - you could either slug it to `my_fancy_github_webhook` with uncertainty of whenever people feel like using dashes or underscores for the service names or you could use these UUIDs which are a lot more handy in URL paths.

Reply:
Oh right! At first I thought we would be using only the `/github` endpoint, and the app would look into the message body to know if the project is known/authorized. Or use the secret. But I suppose that having UUIDs in the URL paths could even allow us to not have a password, a bit like what OpenShift does.